### DISCUSSION TITLE
change path to created invoices

### DISCUSSION THREAD

**Discussion by BeckeBauer** [kimai/kimai#2187] [Questions & Answers]
When I create an invoice, it is automatically be saved under the web folder `var\data\invoices` 
It would be handy to change the invoice folder to another location on the server (were it can be accessed also from outside Kimai). Maybe, it can be realised by an option in local.yaml but until then (if ever) I would appreciate a hint about which / how to change the relevant file(se) to set the path...

**Comment by kevinpapst** [comment]
Very basic fix: create a symlink to the invoices directory like in `ln -s /var/www/kimai2/var/data/invoices/ ~/kimai-invoices/`.

Or you change the `data_dir` key in `local.yaml`. 

This is explained here: https://www.kimai.org/documentation/configurations.html#data-directory

**Comment by BeckeBauer** [comment]
Thank you for the suggestions. 

I have troubles to move the `data_dir`  to a location outside the web folder due to restrictions (open_basedir and if deactivated, it seems that the http_user cannot be given write permissions). 

I will continue to move the invoices forth and back by hand between the intended invoice folder (mapped as network drive) and the Kimai invoice web folder (mapped as another network drive) since a symlink between the two folders does not work if both are mapped as network drives (at least on my windows machine)

**Comment by kevinpapst** [comment]
Nothing of that is a limitation of Kimai 😁 

I have a couple of hundred installations that use this setup, so it is something very basic that you are missing.
open_basedir can be configured to include that path and permissions for system user (like httpd) need to be applied recursively down to the lowest directory. 
Not the write permissions but read and execute (to be able to change into that directory).

**Comment by BeckeBauer** [comment]
You were right, after granting http user permissions down to the lowest directory, the irritation open_basedir errors in the log are gone and Kimai accepts the `data_dir` at another location. 

The follow-up questions are: 
- is there a way to have the `custom-css-bundle.css` stored at another place (so that it does not need to move with the invoices)
- can the invoice folder be split into subfolders named by the year where the invoice was created, e.g. all invoices from 2020 are stored in the subfolder 2020 (or invoices_2020), all invoices from 2021 are stored in the subfolder 2021 (or invoices_2021) and so on?
The way to filter invoices directly in Kimai is not very elaborated (yet) so that a division by folder names would help when browsing the folder for a particular invoice

**Comment by kevinpapst** [comment]
Instead of moving the data/invoices directory you could also create a symlink `ln -s /tmp/invoices/ /var/www/kimai2/var/data/invoices` inside Kimai. 
Might be the easier way, instead of moving the data directory completely.

- I will not add a separate config for the invoice directory, the data directory should contain all directories that store data in Kimai
- Splitting the invoice directories into a structure like `invoices/year/month/` is a nice idea and I will probably add that feature in the future

But probably the more important option is to add filters to the invoice listing page. There is a feature request existing somewhere... vote for it!

**Comment by BeckeBauer** [comment]
I checked the feature requests but could not find a discussion requesting filters for the invoice listings view (there might feature requests that might have it in mind but I am not sure) - anyway, I am quite familiar with links under MS Windows; I am also able to create symlinks on a linux server but I do not quite understand what I achieve with the suggested symlink. 

Let*s say I have the folder "accounting". This is where I moved the data directory. When I map this folder under Windows, the data folder is shown as subfolder and can be accessed. However, when I create a symlink to the data directory, the data folder is shown but cannot be accessed (maybe a Windows permission problem?). But your symlink seems to imply something else?
